src/Symfony/Component/Messenger/Transport/Receiver/ForceConsumptionReceiver.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Receiver;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * @author Mathias STRASSER <contact@roukmoute.fr>
+ *
+ * @experimental in 4.2
+ */
+final class ForceConsumptionReceiver implements ReceiverInterface
+{
+    /** @var ReceiverInterface */
+    private $decoratedReceiver;
+
+    /** @var bool */
+    private $isVerbose;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(ReceiverInterface $decoratedReceiver, bool $isVerbose, LoggerInterface $logger)
+    {
+        $this->decoratedReceiver = $decoratedReceiver;
+        $this->isVerbose = $isVerbose;
+        $this->logger = $logger;
+    }
+
+    public function receive(callable $handler): void
+    {
+        $this->decoratedReceiver->receive(
+            function ($message) use ($handler) {
+                try {
+                    $handler($message);
+                } catch (\Throwable $exception) {
+                    if (!$this->isVerbose) {
```

Comment:
This flag is not useful. If I don't care about the log messages, I can just inject a NullLogger myself.